index.js

```diff
@@ -102,6 +102,20 @@ module.exports.lintFiles = (patterns, opts) => {
 			});
 		}
 
+		const isTSFile = ext => ext === '.ts' || ext === '.tsx';
+
+		if (paths.filter(filePath => isTSFile(path.extname(filePath))).length > 0) {
+			opts.parser = 'typescript-eslint-parser';
+			opts['settings'] = {
```

Comment:
I don't think this is correct as with the current code the entire `settings` value will be overridden.
We want to add/merge only the last leaf (`extensions` and `typescript-eslint-parser`).
So if the user config already contains something like:
```js
{
settings: {
  'import/resolver': {
    node: {extensions: ['.js', '.jsx', '.xyz']}
  }
}
}
```
We want to end up with:
```js
{
settings: {
  'import/resolver': {
    node: {extensions: ['.js', '.jsx', '.xyz', '.ts', '.tsx']}
  }
}
}
```

Reply:
You right